### PR TITLE
VS2017 and VS2022 supported. Bugs fixed.

### PR DESCRIPTION
**Pull request by Beanavil**
Some bugs solved.
VS2017 and VS2022 project files added.

### CODE REVIEW

**Comment by hchevanne** [comment]
- In vulkan_interop_vs2017.sln, vulkan_interop_vs2019.sln and vulkan_interop_vs2022.sln:

Missing "$(GLFW_DIR)\include" in "\<AdditionalIncludeDirectories\>" property for both **HIP_clang** and **HIP_nvcc** toolsets (**Debug** and **Release**)
Missing "<AdditionalLibraryDirectories>$(GLFW_DIR)\lib-**vc2019**</AdditionalLibraryDirectories>" property for both HIP_clang and HIP_nvcc toolsets (Debug and Release)
Note: _**vc2019** should changed according to the target version of VS, (2017, 2019 or 2022)._


- In opengl_interop_vs2017.sln, opengl_interop_vs2019.sln and opengl_interop_vs2022.sln:

Missing "$(GLFW_DIR)\include" in "\<AdditionalIncludeDirectories\>" property for both **HIP_clang** and **HIP_nvcc** toolsets (**Debug** and **Release**)
Missing "<AdditionalLibraryDirectories>$(GLFW_DIR)\lib-**vc2019**</AdditionalLibraryDirectories>" property for both HIP_clang and HIP_nvcc toolsets (Debug and Release)
Note: _**vc2019** should changed according to the target version of VS, (2017, 2019 or 2022)._

**Comment by Snektron** [comment]
@hchevanne The reason that those include/library paths were removed is because these examples are set up to benefit from [vcpkg integrate](https://learn.microsoft.com/en-us/vcpkg/commands/integrate) (as described in the readme). We can add GLFW_DIR usage as a fallback strategy, though.

**Comment by hchevanne** [comment]
This was reported by the CI team.
Also, there are some typos in **Libraries/rocRAND/simple_distributions_cpp** vcxproj files for vs2017, vs2019 and vs2022. 
**-Wno-#warnings** instead of **-Wno-warnings**
In fact, not all example are intended to run on non-AMD hardware. In particular all roc XXXX libraries ones.
Dedicated ROCm-Examples-VS20xx.sln solution files tagged with "HIP_nvcc" should come with only a selection of examples that can be run on non-AMD hardware.

**Comment by Snektron** [comment]
> Also, there are some typos in Libraries/rocRAND/simple_distributions_cpp vcxproj files for vs2017, vs2019 and vs2022.
-Wno-#warnings instead of -Wno-warnings

[clang documentation](https://clang.llvm.org/docs/DiagnosticsReference.html#w-warnings) lists this flag as being `-Wno-#warnings` (`-W#warnings`) and not as `-Wno-warnings`. Indeed, when I swap out the current `-Wno-#warnings` with `-Wno-warnings` I get warnings, so Im not really sure what should be changed here...

**Comment by emankov** [comment]
@Beanavil, @Snektron, hello!

May I ask you to make your PRs `a bit smaller`? Actually, much smaller, at least 10x times smaller. It will save us time, and it is how the open-source world lives. There is no need in putting everything in one basket.

Thanks in advance,
Evgeny (HIP/HIPIFY/HIP-VS dev team)

**Comment by emankov** [comment]
> > Also, there are some typos in Libraries/rocRAND/simple_distributions_cpp vcxproj files for vs2017, vs2019 and vs2022.
> > -Wno-#warnings instead of -Wno-warnings
> 
> [clang documentation](https://clang.llvm.org/docs/DiagnosticsReference.html#w-warnings) lists this flag as being `-Wno-#warnings` (`-W#warnings`) and not as `-Wno-warnings`. Indeed, when I swap out the current `-Wno-#warnings` with `-Wno-warnings` I get warnings, so Im not really sure what should be changed here...

Maybe, these options for the `HIP_nvcc` Platform Toolset should be changed? You are trying to feed `nvcc` with clang's options - that's the point.

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2019.sln [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/assembly_to_executable/assembly_to_executable_vs2019.sln]: A solution file's header should start from `Microsoft Visual Studio Solution File, Format Version XX.YY`, empty lines are unacceptable.

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2019.vcxproj [hunk 2]
```diff
@@ -77,7 +77,7 @@
 
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/assembly_to_executable/assembly_to_executable_vs2019.vcxproj]: Should be an exact value, `16.0` in this particular case. Substitution even if it takes place is incorrect for `vcxproj` file targeting a specific version of VS - 2019.

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.4.33213.308
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "assembly_to_executable_vs2022", "assembly_to_executable_vs2022.vcxproj", "{96AF6231-97E6-4FEF-8132-11897D33E973}"
+EndProject
```
Comment (emankov) [inline HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln:6]: Why solutions for every single project are provided as well? What's the point?

File: Applications/floyd_warshall/floyd_warshall_vs2017.vcxproj [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    <ClInclude Include="..\..\Common\cmdparser.hpp" />
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{3bbda23b-43c0-4b91-8ba8-1cfe8b981184}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>floyd_warshall_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline Applications/floyd_warshall/floyd_warshall_vs2017.vcxproj]: Better to have this `PropertyGroup` above just after `ProjectConfigurations` `ItemGroup`. It is not an error, the reason is to have a unified and standard AMAP representation of all the projects for ease of support, first of all.

File: Applications/floyd_warshall/floyd_warshall_vs2019.vcxproj [hunk 5]
```diff
@@ -18,7 +18,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline Applications/floyd_warshall/floyd_warshall_vs2019.vcxproj]: Should be 16.0

File: Applications/floyd_warshall/floyd_warshall_vs2022.vcxproj [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    <ClInclude Include="..\..\Common\cmdparser.hpp" />
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{015df085-feb3-4c7a-acee-7cffb3c9aff0}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>floyd_warshall_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline Applications/floyd_warshall/floyd_warshall_vs2022.vcxproj]: `VCProjectVersion` should be 17.0, and the `PropertyGroup` `Globals` should be moved up (see the above comment on where).

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2017.vcxproj [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    </CustomBuild>
+
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{f72a2ce8-6391-4929-8688-dd5d1e338773}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>assembly_to_executable_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/assembly_to_executable/assembly_to_executable_vs2017.vcxproj]: Same as above

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.vcxproj [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+    </CustomBuild>
+
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{96af6231-97e6-4fef-8132-11897d33e973}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>assembly_to_executable_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.vcxproj]: Same as above

File: HIP-Basic/bandwidth/bandwidth_vs2019.vcxproj [hunk 9]
```diff
@@ -18,7 +18,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/bandwidth/bandwidth_vs2019.vcxproj]: 16.0

File: HIP-Basic/bit_extract/bit_extract_vs2017.vcxproj [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{e9dc3573-5fa9-4c03-8830-909f71ed34d9}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>bit_extract_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/bit_extract/bit_extract_vs2017.vcxproj]: Same as above

File: HIP-Basic/bit_extract/bit_extract_vs2019.sln [hunk 11]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/bit_extract/bit_extract_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/bit_extract/bit_extract_vs2019.vcxproj [hunk 12]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/bit_extract/bit_extract_vs2019.vcxproj]: 16.0

File: HIP-Basic/bit_extract/bit_extract_vs2022.vcxproj [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{7c53480c-5014-4bd7-8b3e-c45a183fa5df}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>bit_extract_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/bit_extract/bit_extract_vs2022.vcxproj]: Same as above

File: HIP-Basic/cooperative_groups/cooperative_groups_vs2017.vcxproj [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{30739f04-9ef3-4f41-8de4-500bdbfecff9}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>cooperative_groups_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/cooperative_groups/cooperative_groups_vs2017.vcxproj]: Same as above

File: HIP-Basic/cooperative_groups/cooperative_groups_vs2019.sln [hunk 15]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/cooperative_groups/cooperative_groups_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/cooperative_groups/cooperative_groups_vs2019.vcxproj [hunk 16]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/cooperative_groups/cooperative_groups_vs2019.vcxproj]: 16.0

File: HIP-Basic/cooperative_groups/cooperative_groups_vs2022.vcxproj [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{6c44066e-6e5b-4dcb-8af9-4e8d0630a849}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>cooperative_groups_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/cooperative_groups/cooperative_groups_vs2022.vcxproj]: Same as above

File: HIP-Basic/device_globals/device_globals_vs2017.vcxproj [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{f6b4fc89-465a-455a-b1e9-ff4c8689e3f0}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>device_globals_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/device_globals/device_globals_vs2017.vcxproj]: Same as above

File: HIP-Basic/device_globals/device_globals_vs2019.sln [hunk 19]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/device_globals/device_globals_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/device_globals/device_globals_vs2019.vcxproj [hunk 20]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/device_globals/device_globals_vs2019.vcxproj]: 16.0

File: HIP-Basic/device_globals/device_globals_vs2022.vcxproj [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{3c13d642-960a-4f99-84a9-5559bce3b347}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>device_globals_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/device_globals/device_globals_vs2022.vcxproj]: Same as above

File: HIP-Basic/device_query/device_query_vs2017.vcxproj [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{a9e1c716-820c-4b29-b36e-4f0ab057f09f}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>device_query_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/device_query/device_query_vs2017.vcxproj]: Same as above

File: HIP-Basic/device_query/device_query_vs2019.sln [hunk 23]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/device_query/device_query_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/device_query/device_query_vs2019.vcxproj [hunk 24]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/device_query/device_query_vs2019.vcxproj]: 16.0

File: HIP-Basic/device_query/device_query_vs2022.vcxproj [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{428c7c0b-c055-4126-a6d2-1e51344e5d40}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>device_query_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/device_query/device_query_vs2022.vcxproj]: Same as above

File: HIP-Basic/dynamic_shared/dynamic_shared_vs2017.vcxproj [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{2e7bb11b-fe0f-419c-aa07-448a1472b593}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>dynamic_shared_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/dynamic_shared/dynamic_shared_vs2017.vcxproj]: Same as above

File: HIP-Basic/dynamic_shared/dynamic_shared_vs2019.sln [hunk 27]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/dynamic_shared/dynamic_shared_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/dynamic_shared/dynamic_shared_vs2019.vcxproj [hunk 28]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/dynamic_shared/dynamic_shared_vs2019.vcxproj]: 16.0

File: HIP-Basic/dynamic_shared/dynamic_shared_vs2022.vcxproj [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{a1d6c8e8-9e43-4703-a368-39fec450548c}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>dynamic_shared_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/dynamic_shared/dynamic_shared_vs2022.vcxproj]: Same as above

File: HIP-Basic/events/events_vs2017.vcxproj [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{aa9e497a-fede-45a4-a6ce-a8e3ddddeb82}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>events_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/events/events_vs2017.vcxproj]: Same as above

File: HIP-Basic/events/events_vs2019.vcxproj [hunk 31]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/events/events_vs2019.vcxproj]: 16.0

File: HIP-Basic/events/events_vs2019.sln [hunk 32]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/events/events_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/events/events_vs2022.vcxproj [hunk 33]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{df601563-b579-4571-88c9-3ec7312d567f}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>events_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/events/events_vs2022.vcxproj]: Same as above

File: HIP-Basic/gpu_arch/gpu_arch_vs2017.vcxproj [hunk 34]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{99e1a6eb-e9e3-48af-a71b-5fe792550f5d}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>gpu_arch_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/gpu_arch/gpu_arch_vs2017.vcxproj]: Same as above

File: HIP-Basic/gpu_arch/gpu_arch_vs2019.sln [hunk 35]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (emankov) [inline HIP-Basic/gpu_arch/gpu_arch_vs2019.sln]: Sln's header should start with `Microsoft Visual Studio Solution File, Format Version XX.YY`

File: HIP-Basic/gpu_arch/gpu_arch_vs2019.vcxproj [hunk 36]
```diff
@@ -17,7 +17,7 @@
     <ClInclude Include="..\..\Common\example_utils.hpp" />
   </ItemGroup>
   <PropertyGroup Label="Globals">
-    <VCProjectVersion>15.0</VCProjectVersion>
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
```
Comment (emankov) [inline HIP-Basic/gpu_arch/gpu_arch_vs2019.vcxproj]: 16.0

File: HIP-Basic/gpu_arch/gpu_arch_vs2022.vcxproj [hunk 37]
```diff
@@ -0,0 +1,129 @@
...
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{d5ceb4c2-4867-4a5c-ad3a-e025ef3c7c0e}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>gpu_arch_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (emankov) [inline HIP-Basic/gpu_arch/gpu_arch_vs2022.vcxproj]: Same as above. 17.0.

File: HIP-Basic/hello_world/hello_world_vs2017.vcxproj [hunk 38]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{bd725c86-e381-4bdd-b3fc-06a42221bebb}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>hello_world_vs2017</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/hello_world/hello_world_vs2017.vcxproj]: Same as above. 15.0.

File: HIP-Basic/hello_world/hello_world_vs2019.vcxproj [hunk 39]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{5e0e9ab0-b708-481f-9226-dd92c3798341}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>hello_world_vs2019</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/hello_world/hello_world_vs2019.vcxproj]: Same as above. 16.0.

File: HIP-Basic/hello_world/hello_world_vs2022.vcxproj [hunk 40]
```diff
@@ -0,0 +1,129 @@
...
+  <ItemGroup>
+    <ClInclude Include="..\..\Common\example_utils.hpp" />
+  </ItemGroup>
+  <PropertyGroup Label="Globals">
+    <VCProjectVersion>$(MSBuildVersion)</VCProjectVersion>
+    <ProjectGuid>{aa92ef7e-2323-4497-accd-b76fb196c545}</ProjectGuid>
+    <Keyword>Win32Proj</Keyword>
+    <RootNamespace>hello_world_vs2022</RootNamespace>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
+  </PropertyGroup>
```
Comment (emankov) [inline HIP-Basic/hello_world/hello_world_vs2022.vcxproj]: Same as above. 17.0.

**Review by emankov — CHANGES_REQUESTED** [review]
Please, fix the rest of `vcxproj` and `sln` files based on the existing comments.

P.S.
There are other changes awaiting merge, so, let's merge these this week.

Thanks,
Evgeny

**Comment by Beanavil** [comment]
@emankov I'm checking the nvcc documentation, and seems like the proper way of passing this compiler options is with [--compiler-options or -Xcompiler](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#compiler-options-options-xcompiler), so `-Xcompiler "-w"` should work. Could you check if this is the case?

File: HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln (hunk 3 above)
Comment (Beanavil) [reply HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln:6]: AFAIK we were asked to include those solution projects, but I was not in the project when this happened. Also, this allows to have each example as a self-contained one, this has advantages such as being able to build a single example in VScode without needing to build the whole examples project.

Comment (saadrahim) [reply HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln:6]: It is a requirement to have a solution for each example. Each example can must be buildable separately and as part of the entire project as well. This is for end user convinience.

Comment (emankov) [reply HIP-Basic/assembly_to_executable/assembly_to_executable_vs2022.sln:6]: Ok, got it, thanks!

**Comment by emankov** [comment]
> @emankov I'm checking the nvcc documentation, and seems like the proper way of passing this compiler options is with [--compiler-options or -Xcompiler](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#compiler-options-options-xcompiler), so `-Xcompiler "-w"` should work. Could you check if this is the case?

Well, we are talking about different compilers, right? Different compilers - different options. There is no any guaranteed correspondence between them. There are some, which came from the GNU C/C++ world, and that is it.

As for the `-Xcompiler`, it is for passing options to nvcc's compiler's frontend/preprocessor directly, and that is it. I do not observe an exact correspondence between warning-related options in `clang` and in `nvcc`.

Analogues redirecting options exist for clang driver: `-Xpreprocessor`, `-Xclang`, `-Xassembler`, etc...

**Comment by Beanavil** [comment]
You are right, `-Xcompiler` should not be used in this case. However, `-w` (or, equivalently, `--disable-warnings`) alone seems like [does the same](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#generic-tool-options) for nvcc as `-Wno-#warnings` for clang.

**Comment by saadrahim** [comment]
@Beanavil and @Snektron what is left to address @emankov's concerns? Can we merge this soon?

**Review by saadrahim — APPROVED** [review]
Please ensure this is the last PR that is this large. Bug fixes should be a separate PR while feature implementations should come in logical chunks. 477 files are impossible to review.

**Comment by Beanavil** [comment]
@saadrahim the changes proposed by @emankov are ready to be pushed, I just wanted some reply to my last comment, so I can be sure that the mentioned nvcc option works properly on your end (I can't check it myself).

**Review by emankov — APPROVED** [review]
Thanks!

**Pull request merged**